metricbeat/module/prometheus/collector/collector.go

```diff
@@ -49,20 +52,36 @@ func init() {
 // MetricSet for fetching prometheus data
 type MetricSet struct {
 	mb.BaseMetricSet
-	prometheus p.Prometheus
+	prometheus     p.Prometheus
+	includeMetrics []*regexp.Regexp
+	ignoreMetrics  []*regexp.Regexp
 }
 
 // New creates a new metricset
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	config := defaultConfig
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
 	prometheus, err := p.NewPrometheusClient(base)
 	if err != nil {
 		return nil, err
 	}
 
-	return &MetricSet{
+	ms := &MetricSet{
 		BaseMetricSet: base,
 		prometheus:    prometheus,
-	}, nil
+	}
+	ms.ignoreMetrics, err = compilePatternList(config.MetricsFilters.IgnoreMetrics)
+	if err != nil {
+		return nil, err
+	}
+	ms.includeMetrics, err = compilePatternList(config.MetricsFilters.IncludeMetrics)
+	if err != nil {
+		return nil, err
```

Comment:
We may want to add some context to these errors to know if the error is in the include metrics patterns or in the ignore metrics ones.

Reply:
👍